### ISSUE TITLE
get user commits from orgs?

### ISSUE THREAD

**Issue by tipsy** [tipsy/github-profile-summary#16]
ruslan suggests to get all user commits from orgs via `listUserOrgs(user).forEach { org -> org.repos.forEach { repo -> repo.statictics } }` 
this will potentially result in thousands of requests depending on what kind of orgs you're a member of

maybe request limit can be increased if users auths? maybe there is a better way?

**Comment by tipsy** [comment]
To the people who want this: Would you be okay with authorizing the app to let it make requests with your token? I can't think of any way to do this with my own token that won't instantly trigger the rate-limit.

**Comment by mithlesh4257** [comment]
Shall we not first list user organizations and find commits to each orgs ?
https://developer.github.com/v3/orgs/#list-user-organizations
https://developer.github.com/v3/repos/commits/

This may work.

**Comment by tipsy** [comment]
@mithlesh4257 Writing the code for it isn't really a problem, but let's say I'm a member of `google`, which has 1148 repos. I picked ten repos at random and got an average of 800 commits (I have no idea if this number is correct, but let's assume it is for now). 
The max page size for the API is 100, which gives us 1140/100 * 800/100 = 12 * 80  = 960 requests. These requests come on top of the normal user requests, and only account for 1 organization. I'm already approaching the rate-limit every hour, and several people who work for `google`, `microsoft`, and other huge organizations have been using the tool. 

I suppose we could run a batch-job for organizations to cache the relevant information. It would mean slightly outdated info, but that's probably okay. It's a bit of work though.

**Comment by tipsy** [comment]
It seems that it's possible to use the event-api to achieve this: 

```
val repoCommits2 = GhService.getUserEvents(username).filter { event -> event.type == Event.TYPE_PUSH }.map {
    it.repo to (it.payload as PushPayload).commits
}.toMap()
```

**Comment by matejkramny** [comment]
Using the App approach to increase API limits is good, better than the stargaze method

**Comment by tipsy** [comment]
The event API is basically useless:

> Events support pagination, however the per_page option is unsupported. The fixed page size is 30 items. Fetching up to ten pages is supported, for a total of 300 events.

**Comment by tedivm** [comment]
I would definitely be okay with authenticating.

**Comment by aduguid** [comment]
This feature would be awesome!

**Comment by lordadamson** [comment]
any news on this?

**Comment by tipsy** [comment]
@lordadamson no, I've pretty much stopped adding new functionality to this project. If you wanna create a PR I'll merge it though.

**Comment by tipsy** [comment]
Probably. Traffic is pretty low on the project these days, so I think it would also be okay to try without, then if the number of requests falls under a certain limit require auth.

**Comment by dolmen** [comment]
Use case: I have so many repos that I have created organizations for projects per language:
- https://github.com/dolmen
- https://github.com/dolmen-go
- https://github.com/dolmen-perl
So I would like to be able to see https://profile-summary-for-github.com/user/dolmen-go

**Comment by dolmen** [comment]
Note that *Commit per quarter* doesn't work for an organization.

**Comment by tipsy** [comment]
I don't have the time to work on this project at the moment, but pull requests are welcome @dolmen !
